presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardManager.java

```diff
@@ -52,6 +52,25 @@
      */
     void replaceShardUuids(long transactionId, long tableId, List<ColumnInfo> columns, Set<UUID> oldShardUuids, Collection<ShardInfo> newShards, OptionalLong updateTime);
 
+    /**
+     * Replace oldShardsUuids with newShards.
+     * Used by compaction.
+     * With tableSupportsDeltaDelete: Delete oldShardsUuids with their delta shards
+     * Add newShards
+     */
+    void replaceShardUuids(long transactionId, boolean tableSupportsDeltaDelete, long tableId, List<ColumnInfo> columns, Map<UUID, Optional<UUID>> oldShardUuids, Collection<ShardInfo> newShards, OptionalLong updateTime);
```

Comment:
- Move `boolean tableSupportsDeltaDelete` to the end.
- The two `replaceShardUuids` signatures only differs at `boolean tableSupportsDeltaDelete`. Why don't we just remove the above one?
- The only caller of `replaceShardUuids` has tableSupportsDeltaDelete = false. Is that intensional?

Reply:
We actually merged them together once, but we want the original call stay as much same as possible, so that our functionality won't pollute the old code path.
We also left a todo when our release is stable: Will merge these new function with old function once new feature is stable